src/trans/cpu/internal/ledir_mod.F90

```diff
@@ -189,7 +189,7 @@ SUBROUTINE LEDIR(KM,KMLOC,KFC,KIFC,KSL,KDGLU,KLED2,PAIA,PSIA,POA1,PW)
   DO JK=1,KFC,ISKIP
     IFLD=IFLD+1
     DO J=1,KDGLU
-      ZB(J,IFLD)=PSIA(JK,ISL+J-1)*PW(ISL+J-1)
+      ZB(J,IFLD)=PSIA(JK,ISL+J-1)*REAL(PW(ISL+J-1),JPRB)
```

Comment:
Since the weights `F%RW` have been promoted from `JPRB` to `JPRD`, do we have any idea about the cost of the necessary casting here? Could it be significant enough to be a noticeable performance penalty?